controller/array_action/array_connection_manager.py

```diff
@@ -113,7 +116,9 @@ def detect_array_type(self):
         logger.debug("detecting array connection type")
 
         for storage_type, port in [(XIVArrayMediator.array_type, XIVArrayMediator.port),
-                                   (SVCArrayMediator.array_type, SVCArrayMediator.port)]:  # ds8k : 8452
+                                   (DS8KArrayMediator.array_type, DS8KArrayMediator.port),
```

Comment:
This is very easy to break.
We have to have a special unit test for this port 22 case, that is not sensitive to the ordering.
i.e. instead of hardcoded `socket_connet.side_effect = [1, 1, 1, 1, 0]`, we need a smart mock that returns 0 for both 22 and 8452 ports (and asserts DS8K type)

Reply:
Yes, but as I wrote, the ut relies on the current order of `socket_connet` calls of the tested code.
This is bad, because for example, in the future, a code maintainer can easily change the order (and ut accordingly), without knowing about port 22 on ds8k.
But if the ut will explicitly mock ds8k with both ports open, it will inform the maintainer about this delicate situation.

I hope this is clear.

Reply:
Write comment that the order is important and explain why in it
also remove # ds8k : 8452 near svc